server/datastore/mysql/software_installers.go

```diff
@@ -10,6 +10,56 @@ import (
 	"github.com/jmoiron/sqlx"
 )
 
+func (ds *Datastore) ListPendingSoftwareInstallDetails(ctx context.Context, hostID uint) ([]string, error) {
+	const stmt = `
+  SELECT
+    hsi.execution_id
+  FROM
+    host_software_installs hsi
+  WHERE
+    hsi.host_id = ?
+  AND
+    hsi.install_script_exit_code IS NULL
+  AND
+    hsi.pre_install_query_output IS NULL
```

Comment:
Oh just thought about it, it should be returned by `created_at` ascending order, so that first install requests get processed first (AIUI this is the datastore method that returns the install uuid requests for orbit to process). 

I think at some point we'll want to enforce the order across the "unified queue", but I'm not sure if that's in this story, could'nt see it in the epic ticket (@roberto maybe?).

Reply:
good catch, IIRC we don't need to do the unified queue stuff, but good idea to return the entries in that order